packages/order-utils/src/signature_utils.ts

```diff
@@ -144,20 +161,33 @@ export const signatureUtils = {
      * @param   data          The hex encoded data signed by the supplied signature.
      * @param   signature     A hex encoded presigned 0x Protocol signature made up of: [SignatureType.Presigned]
      * @param   signerAddress The hex encoded address that signed the data, producing the supplied signature.
+     * @param   exchangeAddress The address of the Exchange contract to validate the signature against.
      * @return  Whether the data was preSigned by the supplied signerAddress.
      */
     async isValidValidatorSignatureAsync(
         supportedProvider: SupportedProvider,
         data: string,
         signature: string,
         signerAddress: string,
+        exchangeAddress?: string,
     ): Promise<boolean> {
         const provider = providerUtils.standardizeOrThrow(supportedProvider);
         assert.isHexString('data', data);
         assert.isHexString('signature', signature);
         assert.isETHAddressHex('signerAddress', signerAddress);
-        const validatorSignature = parseValidatorSignature(signature);
-        const exchangeContract = new ExchangeContract(signerAddress, provider);
+
+        let exchangeContract: ExchangeContract;
+        if (exchangeAddress) {
```

Comment:
Nit: We prefer to be explicit and avoid using "truthiness"
```suggestion
        if (exchangeAddress !== undefined) {
```

Reply:
Fixed it in 42d5bdd3abd22bea9782606bd3a47e643e5cbeb1.